lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2316,39 +2317,28 @@ void PartitionCloner::visitOpInst(SingleValueInstruction *inst,
   ValueMap[inst] = result;
 }
 
-/// Create the operands a const tensor inst in accelerator code via `B` at
-/// `loc`, based on the literal in `hostLiteralInst`.  Extend `opName` and
-/// `operands` with the attributes of that const tensor inst, which caller will
-/// then use to finish creating the inst.
-// TODO: migrate to the new graph op inst.
-static void createConstTensorAttrsOnAccelLegacy(
-    SingleValueInstruction *hostLiteralInst, SILLocation loc, ASTContext &ctx,
-    SILBuilder &B, std::string &opName, SmallVectorImpl<SILValue> &operands) {
-  // Literals take attributes specifying the dtype and value.
-  opName += ",dtype$dtype,value$tensor";
-
-  auto dtype = convertSwiftTypeToTF(hostLiteralInst->getType().getASTType());
-  auto dtypeCst = B.createIntegerLiteral(
-      loc, SILType::getBuiltinIntegerType(32, ctx), dtype);
-  operands.push_back(dtypeCst);
-
-  // The value attribute is specified by a clone of the literal itself.
-  auto ourCst = hostLiteralInst->clone(dtypeCst);
-  ourCst->setDebugLocation(B.getSILDebugLocation(loc));
-  operands.push_back(ourCst);
-}
-
+/// Create the attributes of a const tensor inst in accelerator code via `B` at
+/// `loc`, where `constVal` and `valTy` respectively specify the value and data
+/// type.  Extend `opName` and `attributes` with the attributes of that const
+/// tensor inst, which caller will then use to finish creating the inst.
+///
+/// For an integer, `constVal` is an APInt. For a float, `constVal` is an
+/// APFloat. See convertValuesToTensor() in graph lowering on how the
+/// constructed const tensor inst get lowered.
 static void createConstTensorAttrsOnAccel(
-    unsigned constVal, SILType valTy, SILLocation loc, ASTContext &ctx,
-    SILBuilder &B, std::string &opName,
+    SymbolicValue constVal, SILType valTy, SILLocation loc,
+    ASTContext &ctx, SILBuilder &B, std::string &opName,
```

Comment:
You can drop the loc, Builder and opName arguments here (yay!)

Reply:
Nice catch. Done!